packages/url_launcher/url_launcher/test/url_launcher_test.dart

```diff
@@ -56,142 +49,128 @@ void main() {
-          'universalLinksOnly': false,
-          'headers': <String, String>{},
-        })
-      ],
+      verify(mock.launch(
+        any,
+        useSafariVC: anyNamed('useSafariVC'),
+        useWebView: captureAnyNamed('useWebView'),
+        enableJavaScript: captureAnyNamed('enableJavaScript'),
+        enableDomStorage: anyNamed('enableDomStorage'),
+        universalLinksOnly: anyNamed('universalLinksOnly'),
+        headers: anyNamed('headers'),
+      )).captured,
+      <bool>[true, true],
     );
   });
 
   test('launch force WebView enable DOM storage', () async {
     await launch('http://example.com/',
         forceWebView: true, enableDomStorage: true);
     expect(
-      log,
-      <Matcher>[
-        isMethodCall('launch', arguments: <String, Object>{
-          'url': 'http://example.com/',
-          'useSafariVC': true,
-          'useWebView': true,
-          'enableJavaScript': false,
-          'enableDomStorage': true,
-          'universalLinksOnly': false,
-          'headers': <String, String>{},
-        })
-      ],
+      verify(mock.launch(
+        any,
+        useSafariVC: anyNamed('useSafariVC'),
+        useWebView: captureAnyNamed('useWebView'),
+        enableJavaScript: anyNamed('enableJavaScript'),
+        enableDomStorage: captureAnyNamed('enableDomStorage'),
+        universalLinksOnly: anyNamed('universalLinksOnly'),
+        headers: anyNamed('headers'),
+      )).captured,
+      <bool>[true, true],
     );
   });
 
   test('launch force SafariVC to false', () async {
     await launch('http://example.com/', forceSafariVC: false);
     expect(
-      log,
-      <Matcher>[
-        isMethodCall('launch', arguments: <String, Object>{
-          'url': 'http://example.com/',
-          'useSafariVC': false,
-          'useWebView': false,
-          'enableJavaScript': false,
-          'enableDomStorage': false,
-          'universalLinksOnly': false,
-          'headers': <String, String>{},
-        })
-      ],
+      // ignore: missing_required_param
+      verify(mock.launch(
+        any,
+        useSafariVC: captureAnyNamed('useSafariVC'),
+        useWebView: anyNamed('useWebView'),
+        enableJavaScript: anyNamed('enableJavaScript'),
+        enableDomStorage: anyNamed('enableDomStorage'),
+        universalLinksOnly: anyNamed('universalLinksOnly'),
+        headers: anyNamed('headers'),
+      )).captured.single,
+      false,
     );
   });
 
   test('cannot launch a non-web in webview', () async {
     expect(() async => await launch('tel:555-555-5555', forceWebView: true),
-        throwsA(isInstanceOf<PlatformException>()));
+        throwsA(isA<PlatformException>()));
   });
 
   test('closeWebView default behavior', () async {
     await closeWebView();
-    expect(
-      log,
-      <Matcher>[isMethodCall('closeWebView', arguments: null)],
-    );
+    verify(mock.closeWebView());
   });
 }
+
+class MockUrlLauncher extends Mock implements UrlLauncherPlatform {}
```

Comment:
This was a red flag for me as I was planning to enforce that it's not implemented with `implements`.
I sent #2230 which leaves a back door for mocks.

Reply:
Done.